10/README.md

```diff
@@ -20,31 +20,114 @@ fast, scalable network applications. Node.js uses an event-driven, non-blocking
+**Accessing the application:**
+```
+$ oc get pods
+$ oc exec <pod> -- curl 127.0.0.1:8080
+```
 
-    ```
-    $ s2i build https://github.com/sclorg/s2i-nodejs-container.git --context-dir=10/test/test-app/ ubi8/nodejs-10 nodejs-sample-app
-    ```
+Source-to-Image framework and scripts
+---------------------
+This image supports the [Source-to-Image](https://docs.openshift.com/container-platform/3.11/creating_images/s2i.html)
+(S2I) strategy in OpenShift. The Source-to-Image is an OpenShift framework
+that makes it easy to write images that take application source code as
+an input, use a builder image like this Node.js container image, and produce
+a new image that runs the assembled application as output.
+
+In order to support the Source-to-Image framework, there are some interesting scripts inside the builder image:
+
+* The `/usr/libexec/s2i/assemble` script inside the image is run in order to produce a new image with the application artifacts. The script takes sources of a given application and places them into appropriate directories inside the image. It utilizes some common patterns in Node.js application development (see **Environment variables** section below).
+* The `/usr/libexec/s2i/run` script executes the application and is set as a default command in the resulting container image.
+
+Build an application using a Dockerfile
+---------------------
+Compared to the Source-to-Image strategy, using a Dockerfile is a more
+flexible way to build an Node.js container image with an application.
+Use it when the Source-to-Image is not flexible enough for you or
+when you build the image outside of the OpenShift environment.
+
+In order to use this image in a Dockerfile, follow these steps:
+
+#### 1. Pull a base builder image to build on
+
+```
+podman pull ubi8/nodejs-10
+```
+
+An UBI image `ubi8/nodejs-10` is used in this example. This image is usable and freely redistributable under the terms of the UBI End User License Agreement (EULA). See more about UBI at [UBI FAQ](https://developers.redhat.com/articles/ubi-faq).
+
+#### 2. Pull an application code
+
+An example application available at https://github.com/sclorg/nodejs-ex.git is used here. Feel free to clone the repository for further experiments.
+
+```
+git clone https://github.com/sclorg/nodejs-ex.git app-src
+```
+
+#### 3. Prepare an application inside a container
+
+This step usually consists of at least these parts:
+
+* putting the application source into the container
+* installing the dependencies
+* setting the default command in the resulting image
+
+For all these three parts, we can either setup all manually and use commands `nodejs` and `npm` explicitly in the Dockerfile (3.1.), or users can use the Source-to-Image scripts inside the image (3.2.; see more about these scripts in the section "Source-to-Image framework and scripts" above), that already know how to set-up and run some common Node.js applications.
```

Comment:
(3.1.) Please use a reference like: https://github.com/hhorak/s2i-nodejs-container/blob/doc-dockerfile/12/README.md#31-to-use-own-setup-create-a-dockerfile-with-this-content

Reply:
The same for 3.2.

Reply:
`We can either` replace with `Users can either`, because of later on in this sentence you have mention `users can use`

Reply:
I'm not sure it is a good idea, this markdown will also be available on container catalog (like here https://catalog.redhat.com/software/containers/detail/5d3fff015a13461f5fb8635a), where links to github anchors don't make sense IMO. I'll try to use some local anchors if they work both on github and on the catalog.

Reply:
Hm, looks like github and catalog generates the same anchor IDs, great! Fixed now.